subiquity/models/subiquity.py

```diff
@@ -278,6 +280,10 @@ def _cloud_init_config(self):
             if getattr(model, 'make_cloudconfig', None):
                 merge_config(config, model.make_cloudconfig())
         merge_cloud_init_config(config, self.userdata)
+        # Raises cloudinit.config.schema.SchemaValidationError or
+        # SchemaDeprecationError to alert to merged-cloud-config issues
```

Comment:
Would it be more consistent with cloud-init to make SchemaDeprecationError non-fatal, and just log it?

Reply:
Sure we can definitely do that as you prefer, ideally I'd love to see the live installer grow some sort of user-facing error dialog handler that would pop up an error message in the UX to inform the person of an actionable problem before they commit the the full install and reboot, or have to spelunk in /var/log/installer/* to see what a hard the traceback/runtime error says.  (maybe that type of error dialog already exists in the installer and I'm not aware of how to plug into it).


 For what it's worth, one thing we found in cloud-init was that warning logs generally aren't discovered until sometime later on a deployment when someone happens to recognize that the setup of the VM didn't quite work in every way they expected. Long-term, we've decided to make cloud-init noisier earlier to avoid delayed awareness of misconfiguration which could lead to extended security holes in a deployed system.  That said, cloud-init is not erroring hard at the moment on schema validation errors because we want to make our schema strict and ensure all unexpected use-cases are resolved. I think it's worth us leaving subiquity in non-strict, warning mode for the moment until cloud-init also applies strict/hard errors on schema validation. We can revisit these warnings in the live installer at that time and maybe we'll also have a mechanism to surface those warnings nicely via UX that we can tie into at that point.